jupyterhub/schema.yaml

```diff
@@ -1080,17 +1080,21 @@ properties:
       existingSecret:
         type: [string, "null"]
         description: |
-          This option allow you to provide the name of an existing k8s Secret to
-          use alongside of the chart managed k8s Secret. The content of this k8s
-          Secret will be merged with the chart managed k8s Secret, giving
-          priority to the self-managed k8s Secret.
+          The name of an existing k8s Secret to use alongside of the chart-
+          managed k8s Secret. This k8s Secret should contain a file,
+          `values.yaml`, which will be merged into the `values.yaml` provided
+          to your chart deployment. The values in this secret will override
+          those from the main chart config file.
 
           ```{warning}
-          1. The self managed k8s Secret must mirror the structure in the chart
+          1. The self-managed k8s Secret must mirror the structure in the chart-
              managed secret.
-          2. [`proxy.secretToken`](schema_proxy.secretToken) (aka.
-             `hub.config.ConfigurableHTTPProxy.auth_token`) is only read from
-             the chart managed k8s Secret.
+
+          2. Several `hub.config` values are only read from the chart-managed
+             k8s secret:
+             - `hub.config.ConfigurableHTTPProxy.auth_token`
+             - `hub.config.JupyterHub.cookie_secret`
+             - `hub.config.CryptKeeper.keys`
```

Comment:
Its specifically the `hub.config.ConfigurableHTTPProxy.auth_token` which is only read from this, rather than the existing secret.

This is because that is the only configuration that is needed outside the `hub` pod where JupyterHub runs, being [mounted here](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/blob/dec9fad9aa97446e276cb8b58e8d9f6e7402f13d/jupyterhub/templates/proxy/deployment.yaml#L117-L124) as a common secret for use between the `hub` pod and the `proxy` pod. This is why that key is unique specifically, and the others aren't.

The `hub` pod running JupyterHub has logic to combine values though using Python code, and thats fine, as long its not required elsewhere as well. For example [here](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/blob/dec9fad9aa97446e276cb8b58e8d9f6e7402f13d/jupyterhub/files/hub/jupyterhub_config.py#L402) a value of relevance is read via [a function, that prioritizes the user managed values if provided](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/blob/dec9fad9aa97446e276cb8b58e8d9f6e7402f13d/jupyterhub/files/hub/z2jh.py#L45-L57).

The special treatment given to the `values.yaml` k8s Secret _key_, representing a helm chart configuration, is done in the jupyterhub pod using [this function](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/blob/main/jupyterhub/files/hub/z2jh.py#L13-L30) to merge values from the user managed secret (hub.existingSecret).